plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs

```diff
@@ -182,12 +185,15 @@ instance HasErrorCode FreeVariableError where
     errorCode  FreeUnique {} = ErrorCode 22
 
 -- | Get the 'Index' corresponding to a given 'Unique'.
-getIndex :: (MonadReader Levels m, AsFreeVariableError e, MonadError e m) => Unique -> m Index
-getIndex u = do
+-- Uses supplied handler for free names (uniques).
+getIndex :: MonadReader Levels m => Unique -> (Unique -> m Index) -> m Index
+getIndex u h = do
     Levels current ls <- ask
     case BM.lookup u ls of
         Just foundlvl -> pure $ levelToIx current foundlvl
-        Nothing       -> throwing _FreeVariableError $ FreeUnique u
+        Nothing       ->
+            -- the return index number should be larger than the current level; there no safety check for it.
```

Comment:
This isn't 100% obvious. It would be a weird thing to do - you'd be mapping free variables to in-scope variables, but maybe there's a reason to do that? Perhaps just explain that consequence here. "This should return an index greater than the current level, otherwise it will map unbound variables to bound variables."